{"url":"https://api.github.com/repos/elastic/elasticsearch/issues/4380","repository_url":"https://api.github.com/repos/elastic/elasticsearch","labels_url":"https://api.github.com/repos/elastic/elasticsearch/issues/4380/labels{/name}","comments_url":"https://api.github.com/repos/elastic/elasticsearch/issues/4380/comments","events_url":"https://api.github.com/repos/elastic/elasticsearch/issues/4380/events","html_url":"https://github.com/elastic/elasticsearch/issues/4380","id":23947897,"node_id":"MDU6SXNzdWUyMzk0Nzg5Nw==","number":4380,"title":"cluster reroute returns riddles on failures :)","user":{"login":"synhershko","id":212252,"node_id":"MDQ6VXNlcjIxMjI1Mg==","avatar_url":"https://avatars2.githubusercontent.com/u/212252?v=4","gravatar_id":"","url":"https://api.github.com/users/synhershko","html_url":"https://github.com/synhershko","followers_url":"https://api.github.com/users/synhershko/followers","following_url":"https://api.github.com/users/synhershko/following{/other_user}","gists_url":"https://api.github.com/users/synhershko/gists{/gist_id}","starred_url":"https://api.github.com/users/synhershko/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/synhershko/subscriptions","organizations_url":"https://api.github.com/users/synhershko/orgs","repos_url":"https://api.github.com/users/synhershko/repos","events_url":"https://api.github.com/users/synhershko/events{/privacy}","received_events_url":"https://api.github.com/users/synhershko/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":{"login":"s1monw","id":973334,"node_id":"MDQ6VXNlcjk3MzMzNA==","avatar_url":"https://avatars0.githubusercontent.com/u/973334?v=4","gravatar_id":"","url":"https://api.github.com/users/s1monw","html_url":"https://github.com/s1monw","followers_url":"https://api.github.com/users/s1monw/followers","following_url":"https://api.github.com/users/s1monw/following{/other_user}","gists_url":"https://api.github.com/users/s1monw/gists{/gist_id}","starred_url":"https://api.github.com/users/s1monw/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/s1monw/subscriptions","organizations_url":"https://api.github.com/users/s1monw/orgs","repos_url":"https://api.github.com/users/s1monw/repos","events_url":"https://api.github.com/users/s1monw/events{/privacy}","received_events_url":"https://api.github.com/users/s1monw/received_events","type":"User","site_admin":false},"assignees":[{"login":"s1monw","id":973334,"node_id":"MDQ6VXNlcjk3MzMzNA==","avatar_url":"https://avatars0.githubusercontent.com/u/973334?v=4","gravatar_id":"","url":"https://api.github.com/users/s1monw","html_url":"https://github.com/s1monw","followers_url":"https://api.github.com/users/s1monw/followers","following_url":"https://api.github.com/users/s1monw/following{/other_user}","gists_url":"https://api.github.com/users/s1monw/gists{/gist_id}","starred_url":"https://api.github.com/users/s1monw/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/s1monw/subscriptions","organizations_url":"https://api.github.com/users/s1monw/orgs","repos_url":"https://api.github.com/users/s1monw/repos","events_url":"https://api.github.com/users/s1monw/events{/privacy}","received_events_url":"https://api.github.com/users/s1monw/received_events","type":"User","site_admin":false}],"milestone":null,"comments":2,"created_at":"2013-12-09T09:07:38Z","updated_at":"2014-03-12T20:36:17Z","closed_at":"2014-03-12T20:36:17Z","author_association":"CONTRIBUTOR","active_lock_reason":null,"body":"When issuing a `_cluster/reroute` command and it fails, its quite impossible to understand why it failed.\n\n```\nitamar@data001:~$ curl -XPOST 'localhost:9200/_cluster/reroute' -d '{ \"commands\" : [ { \"move\" : { \"index\" : \"index1\", \"shard\" : 0, \"from_node\" : \"data006\", \"to_node\" : \"data019\" } }, { \"move\" : { \"index\" : \"index2\", \"shard\" : 0, \"from_node\" : \"data019\", \"to_node\" : \"data006\" } } ] }'\n\n{\"error\":\"RemoteTransportException[[data014][inet[/10.10.5.14:9300]][cluster/reroute]]; nested: ElasticSearchIllegalArgumentException[[move_allocation] can't move [index1][0], from [data006][Lh1y_3zkQWiPjfS7DoSteg][inet[data006/10.10.5.1:9300]]{tag=gen1}, to [data019][12_HJbCyQFGx96iTPEoGxw][inet[data019/10.1.2.2:9300]]{tag=gen1}, since its not allowed, reason: [NO()][YES()][YES()][YES()][YES()][YES()][YES()]]; \",\"status\":400}\n```\n\nThe `[NO()][YES()][YES()][YES()][YES()][YES()][YES()]]` part is quite annoying do decrypt and isn't documented anywhere. I know each is a decision on allocation, rebalance etc but in case the command can't go through we need to know why. Either make this a more descriptive error or have this documented somewhere. I'd go with the former since it avoids docs becoming stale.\n\nAs always, I'll be happy to do this myself if this makes sense to you.\n","closed_by":{"login":"s1monw","id":973334,"node_id":"MDQ6VXNlcjk3MzMzNA==","avatar_url":"https://avatars0.githubusercontent.com/u/973334?v=4","gravatar_id":"","url":"https://api.github.com/users/s1monw","html_url":"https://github.com/s1monw","followers_url":"https://api.github.com/users/s1monw/followers","following_url":"https://api.github.com/users/s1monw/following{/other_user}","gists_url":"https://api.github.com/users/s1monw/gists{/gist_id}","starred_url":"https://api.github.com/users/s1monw/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/s1monw/subscriptions","organizations_url":"https://api.github.com/users/s1monw/orgs","repos_url":"https://api.github.com/users/s1monw/repos","events_url":"https://api.github.com/users/s1monw/events{/privacy}","received_events_url":"https://api.github.com/users/s1monw/received_events","type":"User","site_admin":false},"performed_via_github_app":null}